### ISSUE TITLE
🐛 Download All Candidates from pool table gives an empty file the first time you click

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#8956]
## 🐛 Bug

If you select All on the Pool Candidates table and then click "Download CSV" you will get a file with a header but no data rows. 

If you click again (and on subsequent clicks) you will get the correct file.

## 🦋 Expected Behaviour

It should work the first time!

## 📋 Steps to Reproduce

1. Log in as admin
2. Go to the Pool Candidates table
3. Select All
4. Click Download CSV and check the file, it will only be a header line
5. Repeat step 4. It should contain the data now.

## 📸 Screenshot

Add a screenshot (if possible).

## 🙋‍♀️ Proposed Solution

CSV file is getting generated before the GetSelectedPoolCandidates returns with data.
Any time clicking on download after that query is returned, file is generated properly.
We just need to wait for that query to finish before generating the file.

Only fetch data after the Download CSV button is clicked. Only create the file after the data is fetched. Show a spinner while this is happening.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] Clicking on Download CSV after selecting all candidates should download a file with all rows of data the first time.
- [ ] Same goes for printing
- [ ] Show a spinner while data is being fetched

**Comment by gobyrne** [comment]
Duplicate of #8099 but the proposed solutions differ.

Competing solution proposals:
1. Wait to fetch any data until the download button is pressed, then make sure it's loaded before downloading the file
2. Continue to fetch the data right away, but make sure it's finished loading before starting the file download

#1 seems more elegant to me (and is my slight preference). No need to get request before it's needed.

**Comment by gobyrne** [comment]
@tristan-orourke do you have a preference or is one easier than the other?

Pulling this out of the sprint to pick between the 2 proposed solutions.

**Comment by tristan-orourke** [comment]
I like the first solution as well, "Wait to fetch any data until the download button is pressed, then make sure it's loaded before downloading the file". 

@gobyrne is it okay if we replace the Download button with a spinner while we're doing that data loading? It may take some time if you selected a lot of users.

**Comment by gobyrne** [comment]
Can we just show a spinner without replacing the button?
